vertx-mongo-client/src/main/java/io/vertx/ext/mongo/WatchOptions.java

```diff
@@ -0,0 +1,114 @@
+package io.vertx.ext.mongo;
+
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonObject;
+
+@DataObject
+public class WatchOptions {
+
+  public static final int DEFAULT_MAX_AWAIT_TIME_MS = 1000;
+  public static final String FULL_DOCUMENT_UPDATE_LOOKUP = "updateLookup";
+  public static final int DEFAULT_BATCH_SIZE = 1;
+
+  private JsonObject resumeAfter;
+  private String fullDocument;
+  private int batchSize;
+  private int maxAwaitTimeMS;
+  private JsonObject collation;
+
+  public WatchOptions() {
+    maxAwaitTimeMS = DEFAULT_MAX_AWAIT_TIME_MS;
+    batchSize = DEFAULT_BATCH_SIZE;
+  }
+
+  public WatchOptions(WatchOptions options) {
+    resumeAfter = options.resumeAfter;
+    fullDocument = options.fullDocument;
+    batchSize = options.batchSize;
+    maxAwaitTimeMS = options.maxAwaitTimeMS;
+    collation = options.collation;
+  }
+
+  public WatchOptions(JsonObject options) {
+    resumeAfter = options.getJsonObject("resumeAfter");
```

Comment:
Some magic strings could be extract here

Reply:
I am following the convention from other `DataObject` classes in here, should I make this change?

Reply:
In that case no.